src/DotLiquid/StandardFilters.cs

```diff
@@ -931,6 +931,51 @@ public static IEnumerable Compact(object input)
             ary.RemoveAll(item => item == null);
             return ary;
         }
+
+        /// <summary>
+        /// Creates an array including only the objects with a given property value, or any truthy value by default.
+        /// </summary>
+        /// <param name="input">an array to be filtered</param>
+        /// <param name="propertyName">The name of the property to filter by</param>
+        /// <param name="target_value">Value to retain, if null object containing this property are retained</param>
+        /// <returns></returns>
+        public static IEnumerable Where(IEnumerable input, string propertyName, string target_value = null)

```

Comment:
The Shopify specification is ambiguous but I think it would be correct to interpret the target value as of any type. This profile assumes the target value is a string. But say we filter a by a price property, the target value may be a decimal type instead. e.g.

{% assign ten_bucks_products = products | where: "price", 9.99 %}

There is probably precedence in the code base on what is the pattern to deal with ambiguous types.

Reply:
I think existing filters dealing with ambiguously typed inputs use `object` parameter type then test the type (is string, is int, is double, is DateTimeOffset, etc.)

We have existing object to object equality implementation there, which likely you can call:
https://github.com/dotliquid/dotliquid/blob/ddaa286eb24f46cbe19a00a35ef5ea4987c24a8a/src/DotLiquid/Condition.cs#L116

Reply:
@daviburg, the Condition constructor requires string for left/right, would an acceptable alternative be to call SafeTypeInsensitiveEqual in ObjectExtensionMethods.cs?

Also, is ObjectExtensionMethods.cs a better class for IsTruthy/IsFalsy? I could easily refactor them.

Reply:
Yes ObjectExtensionMethods is the right place to look for variable equality helper. If what exists there doesn't work for the object equality you need, I would add the newly needed variant there.

As the ObjectExtensionMethods existing class is pretty small, and so would be the new ThruthyAndFalsy file, I agree it is good to refactor the new extensions into the existing file.

Reply:
Refactored.